src/impl/icetransport.cpp

```diff
@@ -210,9 +215,17 @@ bool IceTransport::addRemoteCandidate(const Candidate &candidate) {
 	return juice_add_remote_candidate(mAgent.get(), string(candidate).c_str()) >= 0;
 }
 
-void IceTransport::gatherLocalCandidates(string mid) {
+void IceTransport::gatherLocalCandidates(string mid, std::vector<IceServer> additionalIceServers) {
 	mMid = std::move(mid);
 
+	int k = 0;
+	for (auto &server : additionalIceServers) {
+		addTURNServer(server);
+		if (++k >= MAX_TURN_SERVERS_COUNT) {
```

Comment:
The counter should now be a class member otherwise it is not enforced properly anymore.

Reply:
It is now a class member